src/tally_filter.F90

```diff
@@ -887,9 +961,9 @@ subroutine get_next_bin_surface(this, p, estimator, current_bin, next_bin, &
     weight = ERROR_REAL
     if (current_bin == NO_BIN_FOUND) then
       do i = 1, this % n_bins
-        if (p % surface == this % surfaces(i)) then
+        if (abs(p % surface) == this % surfaces(i)) then
           next_bin = i
-          weight = ONE
+          weight = sign(1, p % surface)
```

Comment:
It looks like both arguments need to be of the same type ([see here](https://gcc.gnu.org/onlinedocs/gfortran/SIGN.html)) so in order to use `ONE` he'd need to cast `p % surface`.  Since there is a type cast involved no matter what, I think it's fine to stick with these integer arguments.